core-diagnostics/src/lib.rs

```diff
@@ -0,0 +1,51 @@
+#![no_std]
+
+#[macro_use]
+extern crate alloc;
+
+#[cfg(feature = "std")]
+extern crate std;
+
+pub mod debuginfo;
+mod label;
+mod related;
+pub mod reporting;
+pub mod sync;
+
+pub use miette::{
+    self, Diagnostic, IntoDiagnostic, LabeledSpan, NamedSource, Report, Result, Severity,
+    SourceCode, WrapErr,
+};
+pub use tracing;
+
+pub use self::{
+    label::Label,
+    related::{RelatedError, RelatedLabel},
+};
+
+pub(crate) mod utils {
+    pub use miden_crypto::utils::{
+        ByteReader, ByteWriter, Deserializable, DeserializationError, Serializable,
+    };
+}
```

Comment:
What is the benefit of a pattern like this? Isn't it cleaner to, in this example, just use `miden_crypto::utils` in every file where it is used, as the reader can clearly see where these types are defined?

Reply:
> we're still a long ways away from being able to standardize our serialization around `serde`

This hasn't been a goal so far and actually we've tried to stay away from `serde` as much as possible. I'm fine with revisiting our serialization approach, especially since we'll probably need to move away from winterfell dependencies over the next few months - but we should probably figure out if `serde` is what we want to go with (this is a separate discussion though).

Reply:
Any particular reason to stay away from `serde`? IMO it is by far the most useful serialization library in the ecosystem, and nothing else even comes close. The biggest win is that we can trivially support serializing our types to different formats from a single description, which is of particular value with things like `Word`, `Felt`, `QualifiedProcedureName`, etc., that are used in many crates for different purposes.